lib/Varien/Db/Adapter/Pdo/Mysql.php

```diff
@@ -1590,14 +1590,9 @@ public function loadDdlCache($tableCacheKey, $ddlType)
             return $this->_ddlCache[$ddlType][$tableCacheKey];
         }
 
-        if ($this->_cacheAdapter instanceof Zend_Cache_Core) {
+        if ($this->_cacheAdapter) {
             $cacheId = $this->_getCacheId($tableCacheKey, $ddlType);
-            $data = $this->_cacheAdapter->load($cacheId);
-            if ($data !== false) {
-                $data = unserialize($data, ['allowed_classes' => false]);
-                $this->_ddlCache[$ddlType][$tableCacheKey] = $data;
-            }
```

Comment:
Probably these lines I'd guess

Reply:
Nevermind, I missed your previous comment about it supporting non-strings.

Reply:
;-) I missed the serialize hehehe

Reply:
Revisiting this, same issue as https://github.com/MahoCommerce/maho/pull/165#pullrequestreview-2825997206

I think the issue is that if an attacker finds a way to write a file into `var/cache`, then it could turn into an arbitrary code execution.